### ISSUE TITLE
Bug: allow configure `linterOptions.reportUnusedDisableDirectives` in `@typescript-eslint/rule-tester`

### ISSUE THREAD

**Issue by SukkaW** [typescript-eslint/typescript-eslint#10071]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

rule-tester

### Repro Code

```TypeScript
import { RuleTester } from '@typescript-eslint/rule-tester';

const $tester = new RuleTester({
  languageOptions: {
    parserOptions: {
      ecmaFeatures: { jsx: true },
      project: 'tsconfig.json',
      projectService: true,
      tsconfigRootDir: path.join(__dirname, '..', 'tests', 'fixtures'),
      warnOnUnsupportedTypeScriptVersion: false
    }
  },
  linterOptions: {
    reportUnusedDisableDirectives: false
  }
});
```

### Expected Result

In the repro code I have disabled `reportUnusedDisableDirectives` through `linterOptions`, so `Unused eslint-disable directive (no problems were reported).` should not be reported during the test.

### Actual Result

```
AssertionError: Should have 2 errors but had 3:

  {
    ruleId: null,
    message: 'Unused eslint-disable directive (no problems were reported).',
    line: 1,
    column: 1,
    severity: 1,
    nodeType: null,
    fix: { range: [Array], text: ' ' }
  }
```

### Additional Info

I am building a custom eslint rule that works with eslint comments, some of the test cases have `eslint-disable` comments.

It turns out that `@typescript-eslint/rule-tester` always overwrites `linterOptions.reportUnusedDisableDirectives`:

https://github.com/typescript-eslint/typescript-eslint/blob/94c5484f747be901f1610fc37192714a619bb355/packages/rule-tester/src/RuleTester.ts#L716

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.7.0` |
| `@typescript-eslint/parser`            | `8.7.0` |
| `@typescript-eslint/rule-tester`       | `8.7.0` |
| `@typescript-eslint/scope-manager`     | `8.7.0` |
| `@typescript-eslint/typescript-estree` | `8.7.0` |
| `@typescript-eslint/type-utils`        | `8.7.0` |
| `@typescript-eslint/utils`             | `8.7.0` |
| `TypeScript`                           | `5.5.4` |
| `ESLint`                               | `9.11.1` |
| `node`                                 | `22.8.0` |

**Comment by bradzacher** [comment]
What's the motivation behind disabling this?

**Comment by SukkaW** [comment]
> What's the motivation behind disabling this?

I am building a custom eslint rule that works with eslint comments, some of the cases test against the content of the `eslint-disable` comments. And it is kinda annoying to move a test case to `invalid` with an error `Unused eslint-disable directive` when it would have been a valid test case.

**Comment by JoshuaKGoldberg** [comment]
👍 makes sense to me. Here's a quick code snippet that you can run in an `index.test.js` with `npx mocha index.test.js` to reproduce core ESLint respecting `linterOptions.reportUnusedDisableDirectives`:

```js
const eslint = require("eslint");

const core = new eslint.RuleTester({
  linterOptions: {
    reportUnusedDisableDirectives: true,
  },
});

const rule = {
  create() {
    return {};
  },
};

core.run("", rule, {
  valid: ["// eslint-disable-next-line"],
  invalid: [],
});
```
